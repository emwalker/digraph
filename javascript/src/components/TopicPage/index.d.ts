import { Component } from 'react';
import { LocationType, NodeTypeOf } from 'components/types';
import { TopicPage_query_Query$data as Response } from '__generated__/TopicPage_query_Query.graphql';
import { TopicPage_topic$data as TopicType } from '__generated__/TopicPage_topic.graphql';
declare type ViewType = Response['view'];
declare type TopicChildType = NodeTypeOf<TopicType['children']>;
declare type Props = {
    alerts: Object[];
    location: LocationType;
    orgLogin: string;
    topic: TopicType;
    view: ViewType;
};
declare type State = {};
declare class TopicPage extends Component<Props, State> {
    constructor(props: Props);
    static getDerivedStateFromProps: (nextProps: Props) => {};
    get children(): (NonNullable<{
        readonly __typename: "Link";
        readonly id: string;
        readonly " $fragmentSpreads": import("relay-runtime").FragmentRefs<"Link_link">;
    } | {
        readonly __typename: "Topic";
        readonly id: string;
        readonly " $fragmentSpreads": import("relay-runtime").FragmentRefs<"Topic_topic">;
    } | {
        readonly __typename: "%other";
    }> | null)[];
    get displaySynonyms(): readonly {
        readonly name: string;
    }[];
    get isGuest(): boolean;
    get recentActivityLocation(): LocationType;
    get linksToReviewLocation(): LocationType;
    renderTopicChild: (child: TopicChildType | null) => JSX.Element | null;
    renderAddForm: () => JSX.Element;
    renderHeadingDetail: () => JSX.Element | null;
    renderNotification: () => JSX.Element;
    renderTopicViews: () => JSX.Element;
    render: () => JSX.Element;
}
export declare const UnwrappedTopicPage: typeof TopicPage;
export declare const query: import("react-relay").GraphQLTaggedNode;
declare const _default: import("react-relay").Container<Omit<Props, "relay">>;
export default _default;
