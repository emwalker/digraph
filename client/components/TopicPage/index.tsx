'use client'

import { useSuspenseQuery } from '@apollo/experimental-nextjs-app-support/ssr'
import { Button, Card, Pagination, Title } from '@mantine/core'
import { Page } from '@/components/Page'
import { graphql } from '@/lib/__generated__/gql'
import classes from './index.module.css'
import { TopicsQuery } from '@/lib/__generated__/graphql'

const query = graphql(/* GraphQL */ `query Topics($topicId: ID!) {
  view(repoIds: [""], searchString: "", viewerId: "1234") {
    topic(id: $topicId) {
      children(searchString: "", first: 50) {
        edges {
          node {
            ... on Topic {
              id
              displayName
            }
          }
        }
      }
    }
  }
}`)

type Connection = NonNullable<TopicsQuery['view']['topic']>['children']

function searchResults(conn: Connection) {
  const edges = conn?.edges || []

  return edges.map((edge) => {
    if (edge == null) return null
    const { node } = edge
    if (node == null) return null

    if (node.__typename === 'Topic') {
      const { id, displayName } = node
      return (
        <Card
          key={id}
          component="a"
          href={`/topics/${id}`}
          padding="sm"
          radius="md"
          className={classes.card}>
          {displayName}
        </Card>
      )
    }

    return null
  })
}

type Props = {
  topicId: string,
}

export default function Topics({ topicId }: Props) {
  const { data } = useSuspenseQuery(query, { variables: { topicId } })
  const results = data.view?.topic?.children

  return (
    <Page>
      <div className={classes.top}>
        <Title className={classes.title} order={2}>Topics</Title>

        <Button
          component="a"
          href="/topics/new"
          className={classes.addButton}
          >
            Add
        </Button>
      </div>

      { results && (
        <div className={classes.results}>
        { searchResults(results) }
        </div>
      )}

      <Pagination total={10} value={1} my="sm" />
    </Page>
  )
}
