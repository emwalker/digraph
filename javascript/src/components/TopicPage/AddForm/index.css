.private-repo {
  color: #735c0f;
  background-color: #fffbdd;
}
