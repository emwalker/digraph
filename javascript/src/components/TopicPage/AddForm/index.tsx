import React, { Component } from 'react'
import { createFragmentContainer, graphql } from 'react-relay'
import classNames from 'classnames'

import { AddForm_topic as Topic } from '__generated__/AddForm_topic.graphql'
import { AddForm_viewer as Viewer } from '__generated__/AddForm_viewer.graphql'
import AddTopic from './AddTopic'
import AddLink from './AddLink'
import SelectRepository from './SelectRepository'
import './index.css'

type Props = {
  topic: Topic,
  viewer: Viewer,
}

class AddForm extends Component<Props> {
  get className(): string {
    return classNames(
      'border',
      'rounded-1',
      'px-md-2',
      'px-3',
      'mt-3',
    )
  }

  get isPrivateRepo(): boolean {
    const repo = this.props.viewer.selectedRepository
    if (!repo) return false
    return repo.isPrivate
  }

  get repoSelected(): boolean {
    return !!this.props.viewer.selectedRepository
  }

  renderInputFields = () => (
    <>
      <AddTopic
        disabled={!this.repoSelected}
        topic={this.props.topic}
        viewer={this.props.viewer}
      />
      <AddLink
        disabled={!this.repoSelected}
        topic={this.props.topic}
        viewer={this.props.viewer}
      />
    </>
  )

  get selectRepositoryStyle(): Object {
    const backgroundColor = this.isPrivateRepo ?
      this.props.viewer.selectedRepository?.displayColor :
      'transparent'
    return { backgroundColor }
  }

  render = () => (
    <form className={this.className} style={this.selectRepositoryStyle}>
      <SelectRepository
        viewer={this.props.viewer}
      />
      {this.repoSelected && this.renderInputFields()}
    </form>
  )
}

export default createFragmentContainer(AddForm, {
  viewer: graphql`
    fragment AddForm_viewer on User {
      selectedRepository {
        isPrivate
        displayColor
      }

      ...AddLink_viewer
      ...AddTopic_viewer
      ...SelectRepository_viewer
      ...SelectedRepo_viewer
    }
  `,
  topic: graphql`
    fragment AddForm_topic on Topic {
      ...AddLink_topic
      ...AddTopic_topic
    }
  `,
})
