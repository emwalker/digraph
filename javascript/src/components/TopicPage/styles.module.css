.synonyms {
  color: #444d56;
  margin-top: 3px;
}

.synonym:after {
  content: "ãƒ»";
  margin: 0 3px;
}

.synonym:last-child:after {
  content: '';
  margin: 0 3px;
}
