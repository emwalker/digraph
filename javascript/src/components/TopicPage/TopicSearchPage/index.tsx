import React from 'react'
import { graphql, useFragment } from 'react-relay'
import { isEmpty } from 'ramda'

import Page from 'components/ui/Page'
import Subhead from 'components/ui/Subhead'
import SidebarList from 'components/ui/SidebarList'
import Columns from 'components/ui/Columns'
import LeftColumn from 'components/ui/LeftColumn'
import RightColumn from 'components/ui/RightColumn'
import List from 'components/ui/List'
import Link from 'components/ui/Link'
import Topic from 'components/ui/Topic'
import { liftNodes, NodeTypeOf } from 'components/types'
import {
  TopicSearchPage_viewer$key,
  TopicSearchPage_viewer$data as ViewerType,
} from '__generated__/TopicSearchPage_viewer.graphql'
import {
  TopicSearchPage_topic$key,
  TopicSearchPage_topic$data as TopicType,
} from '__generated__/TopicSearchPage_topic.graphql'

type ParentTopicType = NodeTypeOf<TopicType['displayParentTopics']>
type SearchItemType = NodeTypeOf<TopicType['children']>

type Props = {
  topic: TopicSearchPage_topic$key,
  viewer: TopicSearchPage_viewer$key,
}

const topicFragment = graphql`
  fragment TopicSearchPage_topic on Topic @argumentDefinitions(
    searchString: {type: "String"},
  ) {
    id
    displayName

    displayParentTopics(first: 100) {
      edges {
        node {
          displayName
          id
        }
      }
    }

    children(first: 100, searchString: $searchString) {
      edges {
        node {
          __typename

          ... on Topic {
            id
            ...Topic_topic
          }

          ... on Link {
            id
            ...Link_link
          }
        }
      }
    }
  }
`

const viewerFragment = graphql`
  fragment TopicSearchPage_viewer on User {
    id
    ...Topic_viewer
    ...Link_viewer
  }
`

const renderSearchResultItem = (viewer: ViewerType, item: SearchItemType | null) => {
  if (!item) return null

  if (item.__typename === 'Topic') {
    return (
      <Topic
        key={item.id}
        topic={item}
        viewer={viewer}
      />
    )
  }

  if (item.__typename === 'Link') {
    return (
      <Link
        key={item.id}
        link={item}
        viewer={viewer}
      />
    )
  }

  return null
}

export default function TopicSearchPage(props: Props) {
  const topic = useFragment(topicFragment, props.topic)
  const viewer = useFragment(viewerFragment, props.viewer)

  if (topic == null) return <div>Error parsing route</div>

  const {
    children: searchResults,
    displayName,
    displayParentTopics,
  } = topic
  const items = liftNodes<SearchItemType>(searchResults)

  return (
    <Page>
      <div className="px-3 px-md-6 px-lg-0">
        <Subhead
          heading={displayName}
        />
        <Columns>
          <RightColumn>
            <SidebarList
              items={liftNodes<ParentTopicType>(displayParentTopics)}
              placeholder="There are no parent topics for this topic."
              title="Parent topics"
            />
          </RightColumn>
          <LeftColumn>
            <List
              placeholder="There are no items in this list."
              hasItems={!isEmpty(items)}
            >
              {items.map((item) => renderSearchResultItem(viewer, item))}
            </List>
          </LeftColumn>
        </Columns>
      </div>
    </Page>
  )
}
