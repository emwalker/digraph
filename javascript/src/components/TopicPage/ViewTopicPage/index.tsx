import React, { useCallback } from 'react'
import { graphql, useFragment } from 'react-relay'
import { isEmpty } from 'ramda'
import { Link as FoundLink, Location } from 'found'

import Page from 'components/ui/Page'
import Subhead from 'components/ui/Subhead'
import SidebarList from 'components/ui/SidebarList'
import Columns from 'components/ui/Columns'
import LeftColumn from 'components/ui/LeftColumn'
import RightColumn from 'components/ui/RightColumn'
import List from 'components/ui/List'
import Link from 'components/ui/Link'
import Topic from 'components/ui/Topic'
import RepoOwnership from 'components/ui/RepoOwnership'
import { topicPath } from 'components/helpers'
import { Color, NodeTypeOf, liftNodes } from 'components/types'
import {
  ViewTopicPage_viewer$key,
  ViewTopicPage_viewer$data as ViewerType,
} from '__generated__/ViewTopicPage_viewer.graphql'
import {
  ViewTopicPage_topic$key,
  ViewTopicPage_topic$data as TopicType,
} from '__generated__/ViewTopicPage_topic.graphql'
import AddForm from './AddForm'

type TopicChildType = NodeTypeOf<TopicType['children']>

type Props = {
  location: Location,
  topic: ViewTopicPage_topic$key,
  viewer: ViewTopicPage_viewer$key,
}

const viewerFragment = graphql`
  fragment ViewTopicPage_viewer on User {
    id
    ...AddForm_viewer
  }
`

const topicFragment = graphql`
  fragment ViewTopicPage_topic on Topic
  @argumentDefinitions(searchString: {type: "String!", defaultValue: ""})
  {
    displayName
    id
    showRepoOwnership
    viewerCanUpdate

    repoTopics {
      displayColor
    }

    displaySynonyms {
      name
    }

    displayParentTopics(first: 100) {
      edges {
        node {
          displayName
          id
        }
      }
    }

    children(first: 1000, searchString: $searchString)
    @connection(key: "ViewTopicPage_topic_children") {
      edges {
        node {
          __typename

          ... on Topic {
            id
            ...Topic_topic
          }

          ... on Link {
            id
            ...Link_link
          }
        }
      }
    }

    ...AddForm_topic
  }
`

type TopicChildProps = {
  child: TopicChildType | null,
  viewer: ViewerType,
}

function TopicChild({ child, viewer }: TopicChildProps) {
  if (!child) return null

  if (child.__typename == 'Topic') {
    return (
      <Topic
        topic={child}
        viewerId={viewer.id}
      />
    )
  }

  if (child.__typename == 'Link') {
    return (
      <Link
        link={child}
        viewerId={viewer.id}
      />
    )
  }

  return null
}

function renderRepoOwnership(topic: TopicType) {
  const repoColors = topic.repoTopics.map((repoTopic) => repoTopic.displayColor as Color)

  return (
    <div className="mt-2">
      <RepoOwnership
        showRepoOwnership={topic.showRepoOwnership}
        repoColors={repoColors}
      />
    </div>
  )
}

function headingDetail(topic: TopicType) {
  const length = topic.displaySynonyms.length

  if (length < 2) return <div>{renderRepoOwnership(topic)}</div>

  return (
    <div>
      <div className="displaySynonyms h6">
        {topic.displaySynonyms.map(({ name }) => (
          <span key={name} className="synonym">{name}</span>
        ))}
      </div>

      {renderRepoOwnership(topic)}
    </div>
  )
}

const renderNotification = () => (
  <div className="Box p-3 mt-3">
    You must be
    {' '}
    <a href="/login">signed in</a>
    {' '}
    to add and move topics and links.
  </div>
)

function renderTopicViews(topic: TopicType) {
  const recentActivityLocation = {
    pathname: `${topicPath(topic.id)}/recent`,
    query: {},
    search: '',
    state: {
      itemTitle: topic.displayName,
    },
  }

  return (
    <div className="Box Box--condensed mb-3">
      <div className="Box-header">
        <span className="Box-title">This topic</span>
      </div>
      <ul>
        <li className="Box-row">
          <FoundLink to={recentActivityLocation} className="Box-row-link">
            Recent activity
          </FoundLink>
        </li>
      </ul>
    </div>
  )
}

export default function TopicPage(props: Props) {
  const topic = useFragment(topicFragment, props.topic)
  const viewer = useFragment(viewerFragment, props.viewer)
  const children = liftNodes(topic.children)
  const parentTopics = liftNodes(topic.displayParentTopics)
  const renderHeadingDetail = useCallback(() => headingDetail(topic), [headingDetail, topic])

  const canEdit = topic.viewerCanUpdate

  return (
    <Page>
      <Subhead
        heading={topic.displayName}
        renderHeadingDetail={renderHeadingDetail}
      />
      <Columns>
        <RightColumn>
          <SidebarList
            items={parentTopics}
            placeholder="There are no parent topics for this topic."
            title="Parent topics"
          />
          {renderTopicViews(topic)}
          {canEdit
            ? <AddForm topic={topic} viewer={viewer} />
            : renderNotification()}
        </RightColumn>
        <LeftColumn>
          <List
            placeholder="There are no items in this list."
            hasItems={!isEmpty(children)}
          >
            {children.filter(Boolean).map((child) => {
              const key = (child?.__typename == 'Link' || child?.__typename == 'Topic')
                ? child.id
                : 'client:topicChildren:0'

              return (
                <TopicChild
                  key={key}
                  child={child}
                  viewer={viewer}
                />
              )
            })}
          </List>
        </LeftColumn>
      </Columns>
    </Page>
  )
}
