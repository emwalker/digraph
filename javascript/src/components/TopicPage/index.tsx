import React, { Component } from 'react'
import { graphql, createFragmentContainer } from 'react-relay'
import { isEmpty } from 'ramda'
import { Link as FoundLink } from 'found'

import Page from 'components/ui/Page'
import Subhead from 'components/ui/Subhead'
import SidebarList from 'components/ui/SidebarList'
import Columns from 'components/ui/Columns'
import LeftColumn from 'components/ui/LeftColumn'
import RightColumn from 'components/ui/RightColumn'
import List from 'components/ui/List'
import Link from 'components/ui/Link'
import Topic from 'components/ui/Topic'
import RepoOwnership from 'components/ui/RepoOwnership'
import { topicPath } from 'components/helpers'
import { Color, LocationType, NodeTypeOf, liftNodes } from 'components/types'
import { TopicPage_query_Query$data as Response } from '__generated__/TopicPage_query_Query.graphql'
import { TopicPage_topic$data as TopicType } from '__generated__/TopicPage_topic.graphql'
import AddForm from './AddForm'

type ViewType = Response['view']
type TopicChildType = NodeTypeOf<TopicType['children']>
type ParentTopicType = NodeTypeOf<TopicType['displayParentTopics']>

type Props = {
  // eslint-disable-next-line react/no-unused-prop-types
  alerts: Object[],
  location: LocationType,
  orgLogin: string,
  topic: TopicType,
  view: ViewType,
}

type State = {}

class TopicPage extends Component<Props, State> {
  constructor(props: Props) {
    super(props)
    this.state = {}
  }

  static getDerivedStateFromProps = (nextProps: Props) => {
    const shouldAppend = window.flashMessages && nextProps.alerts && nextProps.alerts.length > 0
    if (shouldAppend && window.flashMessages?.addMessage)
      // eslint-disable-next-line nonblock-statement-body-position
      nextProps.alerts.forEach(window.flashMessages.addMessage)

    return {}
  }

  get children() {
    return liftNodes<TopicChildType>(this.props.topic.children)
  }

  get displaySynonyms() {
    return this.props.topic.displaySynonyms
  }

  get isGuest(): boolean {
    return this.props.view.viewer.isGuest
  }

  get recentActivityLocation(): LocationType {
    return {
      pathname: `${topicPath(this.props.topic.id)}/recent`,
      query: {},
      search: '',
      state: {
        itemTitle: this.props.topic.displayName,
      },
    }
  }

  get linksToReviewLocation(): LocationType {
    return {
      pathname: `${topicPath(this.props.topic.id)}/review`,
      query: {},
      search: '',
      state: {
        itemTitle: this.props.topic.displayName,
      },
    }
  }

  renderTopicChild = (child: TopicChildType | null) => {
    if (!child) return null

    if (child.__typename == 'Topic') {
      return (
        <Topic
          key={child.id}
          topic={child}
        />
      )
    }

    if (child.__typename == 'Link') {
      return (
        <Link
          key={child.id}
          link={child}
          viewer={this.props.view.viewer}
        />
      )
    }

    return null
  }

  renderAddForm = () => (
    <AddForm topic={this.props.topic} viewer={this.props.view.viewer} />
  )

  get repoColors(): Color[] {
    return this.props.topic.repoTopics.map((repoTopic) => repoTopic.displayColor as Color)
  }

  renderRepoOwnership = () => 
    <RepoOwnership
      showRepoOwnership={this.props.topic.showRepoOwnership}
      repoColors={this.repoColors}
    />

  renderHeadingDetail = () => {
    const { displaySynonyms } = this
    const { length } = displaySynonyms

    if (length < 2) return <div>{ this.renderRepoOwnership() }</div>

    return (
      <div>
        <div className="displaySynonyms h6">
          {displaySynonyms.map(({ name }) => (
            <span key={name} className="synonym">{name}</span>
          ))}
        </div>

        { this.renderRepoOwnership() }
      </div>
    )
  }

  renderNotification = () => (
    <div className="Box p-3 mt-3">
      You must be
      {' '}
      <a href="/login">signed in</a>
      {' '}
      to add and move topics and links.
    </div>
  )

  renderTopicViews = () => (
    <div className="Box Box--condensed mb-3">
      <div className="Box-header">
        <span className="Box-title">This topic</span>
      </div>
      <ul>
        <li className="Box-row">
          <FoundLink to={this.recentActivityLocation} className="Box-row-link">
            Recent activity
          </FoundLink>
        </li>
        {!this.isGuest && (
          <li className="Box-row">
            <FoundLink to={this.linksToReviewLocation} className="Box-row-link">
              Links to review
            </FoundLink>
          </li>
        )}
      </ul>
    </div>
  )

  render = () => {
    const { location, topic } = this.props

    if (!topic) {
      return (
        <div>
          Topic not found:
          {location.pathname}
        </div>
      )
    }

    const { displayName, displayParentTopics } = topic
    const children = this.children.filter(Boolean)

    return (
      <Page>
        <Subhead
          heading={displayName}
          renderHeadingDetail={this.renderHeadingDetail}
        />
        <Columns>
          <RightColumn>
            <SidebarList
              items={liftNodes<ParentTopicType>(displayParentTopics)}
              placeholder="There are no parent topics for this topic."
              title="Parent topics"
            />
            {this.renderTopicViews()}
            {this.isGuest
              ? this.renderNotification()
              : this.renderAddForm()}
          </RightColumn>
          <LeftColumn>
            <List
              placeholder="There are no items in this list."
              hasItems={!isEmpty(children)}
            >
              {children.map(this.renderTopicChild)}
            </List>
          </LeftColumn>
        </Columns>
      </Page>
    )
  }
}

export const UnwrappedTopicPage = TopicPage

export const query = graphql`
query TopicPage_query_Query(
  $viewerId: ID!,
  $repoIds: [ID!],
  $topicId: String!,
  $searchString: String,
) {
  alerts {
    id
    text
    type
  }

  view(
    viewerId: $viewerId,
    repositoryIds: $repoIds,
  ) {
    viewer {
      id
      isGuest
      ...AddForm_viewer
      ...Link_viewer
    }

    topic(id: $topicId) {
      ...TopicPage_topic @arguments(searchString: $searchString)
    }
  }
}`

export default createFragmentContainer(TopicPage, {
  topic: graphql`
    fragment TopicPage_topic on Topic @argumentDefinitions(
      searchString: {type: "String", defaultValue: ""},
    ) {
      displayName
      id
      showRepoOwnership

      repoTopics {
        displayColor
      }

      displaySynonyms {
        name
      }

      displayParentTopics(first: 100) {
        edges {
          node {
            displayName
            id
          }
        }
      }

      children(first: 1000, searchString: $searchString) @connection(key: "Topic_children") {
        edges {
          node {
            __typename

            ... on Topic {
              id
              ...Topic_topic
            }

            ... on Link {
              id
              ...Link_link
            }
          }
        }
      }

      ...AddForm_topic
    }
  `,
})
