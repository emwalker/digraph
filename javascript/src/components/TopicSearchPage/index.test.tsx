import React from 'react'
import { shallow } from 'enzyme'

import TopicPage from './index'

jest.mock('react-relay', () => ({
  createFragmentContainer: (component: any) => component,
  createRefetchContainer: (Component: any) => (props: any) => (
    <Component {...props} relay={{ refetch: () => {} }} />
  ),
  QueryRenderer: () => null,
}))

describe('<TopicPage />', () => {
  const topic = {
    name: 'Frotz',
  }

  const view = {
    currentRepository: {
      displayName: 'Private collection',
      isPrivate: true,
    },
  }

  // @ts-expect-error
  const wrapper = shallow(<TopicPage topic={topic} view={view} />)

  it('renders', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
