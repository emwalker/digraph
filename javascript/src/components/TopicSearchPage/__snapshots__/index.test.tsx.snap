// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopicPage /> renders 1`] = `
<Page>
  <div
    className="px-3 px-md-6 px-lg-0"
  >
    <Breadcrumbs
      repository={
        Object {
          "displayName": "Private collection",
          "isPrivate": true,
        }
      }
    />
    <Subhead
      heading="Frotz"
      renderHeadingDetail={null}
    />
    <Columns>
      <RightColumn>
        <_default
          items={Array []}
          placeholder="There are no parent topics for this topic."
          repoName="Private collection"
          title="Parent topics"
        />
      </RightColumn>
      <LeftColumn>
        <_default
          hasItems={false}
          placeholder="There are no items in this list."
        />
      </LeftColumn>
    </Columns>
  </div>
</Page>
`;
